ver/version.go

```diff
@@ -43,3 +44,21 @@ func versionString(version, sha string, release, dirty bool) string {
 	}
 	return fmt.Sprintf("%s+%s+dirty", version, sha)
 }
+
+func ParseVersion(version string) (semver.Version, string, bool) {
+	var dirty bool
+	var sha string
+	v := version
+	if strings.HasSuffix(v, ".dirty") {
+		dirty = true
+		v = strings.TrimSuffix(v, ".dirty")
+	}
+	if strings.Contains(v, "-") {
+		tmp := strings.Split(v, "-")
+		v = tmp[0]
+		sha = tmp[1]
+	}
+
+	semVersion, _ := semver.Parse(v)
```

Comment:
why ignore error?

Reply:
good question, this was copied as-is from fogg

Reply:
ah it appears we currently don't do any error handling